packages/server/src/insights/insight-classes.ts

```diff
@@ -224,10 +226,51 @@ export class AverageWaitTime implements InsightInterface<QuestionModel> {
   }
 }
 
+export class AverageHelpingTime extends AverageWaitTime {
+  displayName = 'Avg Helping Time';
+  description = 'Gets the average helping time';
+
+  async compute(filters): Promise<SimpleDisplayOutputType> {
+    const helpTime = await addFilters(
+      createQueryBuilder(QuestionModel)
+        .select(
+          'EXTRACT(EPOCH FROM AVG(QuestionModel.closedAt - QuestionModel.helpedAt)::INTERVAL)/60',
+          'avgHelpTimeInMinutes',
+        )
+        .where(
+          'QuestionModel.helpedAt IS NOT NULL AND QuestionModel.closedAt IS NOT NULL',
+        ),
+      QuestionModel.name,
+      filters,
+      this.possibleFilters,
+    ).getRawOne();
+    return `${Math.floor(helpTime.avgHelpTimeInMinutes)} min`;
+  }
+}
+
+export class QuestionToStudentRatio implements InsightInterface<QuestionModel> {
+  displayName = 'Questions per Student';
+  description = 'How many questions were asked per student on average?';
+  roles = [Role.PROFESSOR];
+  component = InsightDisplay.SimpleDisplay;
+  possibleFilters = ['courseId', 'timeframe'];
+  size = 'small' as const;
+
+  async compute(filters): Promise<SimpleDisplayOutputType> {
+    const totalQuestions = await new TotalQuestionsAsked().compute(filters);
+    const totalStudents = await new TotalStudents().compute(filters);
```

Comment:
Higher Order Insight 😍 

Since each of our insight classes is a singleton we can use the created instance from the `INSIGHTS_MAP` instead of creating a new instance. 

```suggestion
    const totalQuestions = await INSIGHTS_MAP[TotalQuestionsAsked.name].compute(filters);
    const totalStudents = await INSIGHTS_MAP[TotalStudents.name].compute(filters);
```

Reply:
Now that I think more about this, we should probably make `compute` a static function but that will need a little refactoring which can be done in a future PR. Feel free to leave it as is.